makefiles/features_precheck.inc.mk

```diff
@@ -0,0 +1,17 @@
+# Check if provided features are available
+
+ifneq (,$(filter picolibc,$(FEATURES_PROVIDED)) $(filter newlib,$(FEATURES_PROVIDED)))
+  ifneq (,$(filter picolibc,$(FEATURES_PROVIDED)))
+    # Test if picolibc.specs is available
+    ifneq ($(shell $(LINK) -specs=picolibc.specs -E - 2>/dev/null >/dev/null </dev/null ; echo $$?),0)
+      FEATURES_PROVIDED := $(filter-out picolibc,$(FEATURES_PROVIDED))
```

Comment:
Maybe just use `FEATURES_BLACKLIST`? That way it would be more obvious that the feature is not provided due to the setup, rather than due to missing upstream support.

Reply:
seem reasonable i test if it keeps working and change

Reply:
BLACKLISTING does not work as intended

Reply:
fixed that

Reply:
as i just saw you also fixed the Blacklist in #15973